File: perf/cases/python/word_count.py
#!/usr/bin/env python3
from __future__ import annotations

import sys

MOD = 1_000_000_007


def main() -> None:
    n = int(sys.argv[1]) if len(sys.argv) > 1 else 500_000
    counts: dict[str, int] = {}
    for i in range(n):
        word = f"w{(i * 1315423911) & 4095}"
        counts[word] = counts.get(word, 0) + 1

    checksum = 0
    for word, count in counts.items():
        checksum = (checksum + len(word) * count + count * count) % MOD
    print(checksum)


if __name__ == "__main__":
    main()

File: perf/cases/cpp/word_count.cpp
#include <cstdint>
#include <cstdlib>
#include <iostream>
#include <string>
#include <unordered_map>

int main(int argc, char** argv) {
    const std::int64_t n = argc > 1 ? std::atoll(argv[1]) : 500000LL;
    constexpr std::int64_t mod = 1000000007LL;
    std::unordered_map<std::string, std::int64_t> counts;
    counts.reserve(4096);
    for (std::int64_t i = 0; i < n; ++i) {
        std::string word = "w" + std::to_string((i * 1315423911LL) & 4095LL);
        ++counts[word];
    }

    std::int64_t checksum = 0;
    for (const auto& [word, count] : counts) {
        checksum = (checksum + static_cast<std::int64_t>(word.size()) * count + count * count) % mod;
    }
    std::cout << checksum << '\n';
    return 0;
}
